### ISSUE TITLE
Plotting parallel slopes model

### ISSUE THREAD

**Issue by rudeboybert** [moderndive/moderndive#26]
When considering multiple regression with one numerical and one categorical predictor, the `ggplot` code to plot the interaction model is easy while the code to plot the "parallel slopes" model is too hard. Consider adding a `geom_parallel_slopes()` ggplot2 geometry extension to `moderndive` package that behaves like `geom_smooth(method = "lm", ...)`

## Example: Seattle house prices

``` r
suppressPackageStartupMessages(library(tidyverse))
library(knitr)
library(moderndive)

# log10() transformations
house_prices <- house_prices %>%
  mutate(
    log10_price = log10(price),
    log10_size = log10(sqft_living)
  )
```

## Default plot is based on interaction model

Default plot when using col = condition is interaction model: different slopes and intercepts

``` r
ggplot(house_prices, aes(x = log10_size, y = log10_price, col = condition)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm", se = FALSE, size = 1) +
  labs(x = "log10 square feet living space", y = "log10 price in USD", title = "House prices in Seattle: Interaction model")
```

![](https://i.imgur.com/4lnwq0s.png)

Corresponding slopes and intercepts
``` r
lm(log10_price ~ log10_size * condition, data = house_prices) %>% 
  get_regression_table()
#> # A tibble: 10 x 7
#>    term             estimate std_error statistic p_value lower_ci upper_ci
#>    <chr>               <dbl>     <dbl>     <dbl>   <dbl>    <dbl>    <dbl>
#>  1 intercept           3.33      0.451     7.38    0        2.45     4.22 
#>  2 log10_size          0.69      0.148     4.65    0        0.399    0.98 
#>  3 condition2          0.047     0.498     0.094   0.925   -0.93     1.02 
#>  4 condition3         -0.367     0.452    -0.812   0.417   -1.25     0.519
#>  5 condition4         -0.398     0.453    -0.879   0.38    -1.29     0.49 
#>  6 condition5         -0.883     0.457    -1.93    0.053   -1.78     0.013
#>  7 log10_size:cond…   -0.024     0.163    -0.148   0.882   -0.344    0.295
#>  8 log10_size:cond…    0.133     0.148     0.893   0.372   -0.158    0.424
#>  9 log10_size:cond…    0.146     0.149     0.979   0.328   -0.146    0.437
#> 10 log10_size:cond…    0.31      0.15      2.07    0.039    0.016    0.604
```

## Parallel slopes is too hard

Compute all regression points first

``` r
model_price_3_points <-
  house_prices %>%
  lm(log10_price ~ log10_size + condition, data = .) %>%
  get_regression_points()
```
Display a random sample of these regression points
```r
model_price_3_points %>% 
  group_by(condition) %>% 
  sample_n(2)
#> # A tibble: 10 x 6
#> # Groups:   condition [5]
#>       ID log10_price log10_size condition log10_price_hat residual
#>    <int>       <dbl>      <dbl> <fct>               <dbl>    <dbl>
#>  1  7377        5.47       2.86 1                    5.27    0.196
#>  2 15372        5.82       3.00 1                    5.40    0.421
#>  3  9277        5.52       3.00 2                    5.35    0.166
#>  4  5859        5.34       2.96 2                    5.32    0.022
#>  5 19748        5.80       3.38 3                    5.74    0.057
#>  6 13959        5.74       3.28 3                    5.66    0.077
#>  7 17844        5.48       3.02 4                    5.45    0.024
#>  8  5380        5.89       3.36 4                    5.74    0.148
#>  9 18642        5.63       3.54 5                    5.94   -0.305
#> 10  7422        5.69       3.01 5                    5.50    0.189
```

Very cumbersome code to plot
```r
ggplot(house_prices, aes(x = log10_size, y = log10_price, col = condition)) +
  geom_point(alpha = 0.1) +
  labs(y = "log10 price", x = "log10 square footage", title = "House prices in Seattle: Parallel slopes model") +
  geom_line(data = model_price_3_points, aes(y = log10_price_hat), size = 1.5, show.legend = FALSE) +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))
```

![](https://i.imgur.com/RUKXy8w.png)

Corresponding slope and intercepts
``` r
lm(log10_price ~ log10_size + condition, data = house_prices) %>% 
  get_regression_table()
#> # A tibble: 6 x 7
#>   term       estimate std_error statistic p_value lower_ci upper_ci
#>   <chr>         <dbl>     <dbl>     <dbl>   <dbl>    <dbl>    <dbl>
#> 1 intercept     2.88      0.036     80.0    0        2.81     2.95 
#> 2 log10_size    0.837     0.006    134.     0        0.825    0.85 
#> 3 condition2   -0.039     0.033     -1.16   0.246   -0.104    0.027
#> 4 condition3    0.032     0.031      1.04   0.3     -0.028    0.092
#> 5 condition4    0.044     0.031      1.42   0.155   -0.017    0.104
#> 6 condition5    0.096     0.031      3.09   0.002    0.035    0.156
```

Created on 2018-07-08 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0.9000).

**Comment by rudeboybert** [comment]
#33 is inelegant, but gets the job done for now. Can definitely be improved using these [instructions](https://cran.r-project.org/web/packages/ggplot2/vignettes/extending-ggplot2.html) on extending ggplot.

**Comment by ismayc** [comment]
Whoofta. This is tough. I'm still working my way through https://ggplot2.tidyverse.org/articles/extending-ggplot2.html. I'll continue working on it when I can.

**Comment by rudeboybert** [comment]
thankfully we have a stop-gap solution from #33, so no hurry. Also, if this is looking wack, we can also consider just punting.

**Comment by rudeboybert** [comment]
Implemented a temporary solution in #33 with `gg_parallel_slopes()`, will revisit defining a new `geom_` later.

``` r
library(ggplot2)
library(dplyr)
library(moderndive)

# log10() transformations
house_prices <- house_prices %>%
  mutate(
    log10_price = log10(price),
    log10_size = log10(sqft_living)
  )

# Output parallel slopes model plot:
gg_parallel_slopes(y = "log10_price", num_x = "log10_size", cat_x = "condition",
                   data = house_prices, alpha = 0.1) +
  labs(x = "log10 square feet living space", y = "log10 price in USD",
       title = "House prices in Seattle: Parallel slopes model")
```

![](https://i.imgur.com/YmyFz3d.png)

<sup>Created on 2019-03-14 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
